react/features/youtube-player/middleware.js

```diff
@@ -0,0 +1,166 @@
+// @flow
+
+import { getCurrentConference } from '../base/conference';
+import {
+    PARTICIPANT_LEFT,
+    participantJoined,
+    pinParticipant,
+    participantLeft,
+    getLocalParticipant } from '../base/participants';
+import { MiddlewareRegistry, StateListenerRegistry } from '../base/redux';
+
+import { TOGGLE_SHARED_VIDEO, SET_SHARED_VIDEO_STATUS } from './actionTypes';
+import { setSharedVideoStatus, showEnterVideoLinkPrompt, setSharedVideoOwner } from './actions';
+
+const SHARED_VIDEO = 'shared-video';
+
+/**
+ * Middleware that captures actions related to YouTube video sharing and updates
+ * components not hooked into redux.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+// eslint-disable-next-line no-unused-vars
+MiddlewareRegistry.register(store => next => action => {
+    const state = store.getState();
```

Comment:
since you use getState and dispatch here, destructure them in the function definition